grails-app/services/org/pih/warehouse/inboundSortation/CrossDockingStrategy.groovy

```diff
@@ -0,0 +1,39 @@
+package org.pih.warehouse.inboundSortation
+
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.requisition.DemandTypeCode
+import org.pih.warehouse.requisition.OutboundDemand
+
+class CrossDockingStrategy implements PutawayStrategy {
+
+    def demandService
+
+    @Override
+    List<PutawayResult> execute(PutawayContext context, List<Location> locations, Integer quantityRemaining) {
+        Map<DemandTypeCode, List<OutboundDemand>> outboundDemandsMap = demandService.getDemandMap()
+
+        List<PutawayResult> putawayTasks = []
+        for (entry in outboundDemandsMap.entrySet()) {
+            DemandTypeCode demandTypeCode = entry.key
+            List<OutboundDemand> demandsForType = entry.value
+
+            boolean isProductDemanded = demandsForType.any { it.product == context.product }
+
+            if (isProductDemanded) {
+                Location putawayLocation = locations.find { it.locationType?.name?.equalsIgnoreCase(demandTypeCode.code) }
```

Comment:
So this will return the first one that matches the name, but we should probably be looking for all locations with the prefix i.e. WC-00, WC-01, WC-02, sorting them (alphabetically for now) and then returning the first one.